cmd/kube-proxy/app/options.go

```diff
@@ -227,6 +232,13 @@ func (o *Options) Complete(fs *pflag.FlagSet) error {
 	return utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates)
 }
 
+// processV1Alpha1Flags processes v1alpha1 flags which can't be directly mapped to internal config.
+func (o *Options) processV1Alpha1Flags(fs *pflag.FlagSet) {
+	if fs.Changed("bind-address") {
+		o.config.NodeIPOverride = []string{o.bindAddress}
```

Comment:
So I'm not sure that `--bind-address X` and `nodeIPOveride: [X]` should mean the same thing...

In particular, `--bind-address X` means "I want X to be the primary IP, but maybe there's also another IP". But with `nodeIPOverride`, it seems weird to have a single IP imply "but maybe there's also another IP". The whole point of it being an array is so you can specify both, if you want both. So fundamentally, `nodeIPOverride: [X]` feels to me like it should means "X is the only IP". (And, for comparison, that's how it works with `kubelet --node-ip`; we autodetect 1 or 2 IPs if you specify nothing, but if you say `kubelet --node-ip 192.168.1.2`, that means _just_ `192.168.1.2`; you have to specify both IPs explicitly if you want non-autodetected dual-stack.)

But in turn, that implies that you can't just convert `--bind-address` to `nodeIPOverride`... but also, we don't want there to be functionality available via the old API that's not available via the new API...

So, OK, two possibilities:
1. You can say something like `nodeIPOverride: ["192.168.1.2", "::"]` to mean "the node IPs are 192.168.1.2 and whatever IPv6 IP you want"
2. We add an `ipFamilyPolicy` field like in `Service`, which defaults to `PreferDualStack` if you don't specify `nodeIPOverride`, `SingleStack` if you specify a single `nodeIPOverride` value, and `RequireDualStack` if you specify 2 `nodeIPOverride` values. And then `--bind-address 192.168.1.2` would be translated to `nodeIPOverride: ["192.168.1.2"], ipFamilyPolicy: PreferDualStack`

I think I like the latter better.

(But I think we need to think about this for longer than 8 hours, so this isn't making it into 1.31.)